### PR TITLE
Update to Ruby 3.2.2

### PR DESCRIPTION
**Pull request by rwaffen**
Fixes: #148

- update docker ci build script for getting versions
- update all refs for ruby version
- update vagrant image
- regen Gemfile.lock

### CODE REVIEW

File: app/models/data_file.rb [hunk 1]
```diff
@@ -13,15 +13,15 @@ def self.search(environment, key)
     result = {}
     HieraData.new(environment.name).files_including(key.name).each do |file|
       hierarchies[file[:hierarchy_name]] ||= Hierarchy.new(
-        environment: environment,
+        environment:,
         name: file[:hierarchy_name],
```
Comment (tuxmea) [inline app/models/data_file.rb:17]: Kommt das von Rubocop?

Comment (rwaffen) [reply app/models/data_file.rb:17]: yes, this is from rubocop. its seems one does this now this way. see https://dev.to/baweaver/ruby-3-1-shorthand-hash-syntax-first-impressions-19op

**Review by tuxmea — APPROVED** [review]
Please let @oneiros confirm, too

**Review by oneiros — APPROVED** [review]

File: Gemfile [hunk 2]
```diff
@@ -64,6 +64,7 @@ group :linter do
   gem 'rubocop'
   gem 'rubocop-rails'
   gem 'rubocop-rake'
+  gem 'rubocop-capybara'
```
[the remark this answers is not in the document]
Comment (rwaffen) [reply Gemfile:67]: this was no accident, rubocop explicitly wanted that 🤷

**Pull request merged**